cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+)
+
+func ValidateDiscovery(c *kubeadm.Discovery) error {
+	if (c.Token == nil) == (c.File == nil) == (c.HTTPS == nil) {
```

Comment:
`false == false == true => true` actually. This fails only for `true == true == true -> false` I think the correct expression is actually `(a == b == c) != (a && b && c)` but I'll just count the number of nils :)